### ISSUE TITLE
Standalone image: "Connecting to Director localhost:9101, Exitcode=>1"

### ISSUE THREAD

**Issue by UltraBlackLinux** [bacularis/bacularis-docker#5]
Hey there,
I just downloaded the standalone docker image from the docker hub in the hopes of not having to set up a ton manually, but I'm running into an issue. Whenever I open the WebUI it shortly after shows
```
Error code: 4

Message: Problem with connection to bconsole. Output=>Connecting to Director localhost:9101, Exitcode=>1
```
Clicking on the "Director" menu item in the sidebar, I'm running into the same issue, but the webui suggests running the first install wizard, but even after running through that (I found nothing that needed to be changed), I'm still running into the same issue.
Any idea what I'm doing wrong? I found nothing about that on the wiki.
Thanks!

Edit: After running through the API wizard, nothing changed at all. I then decided to manually check the API settings menu and it's now working surprisingly. How does that work?

**Comment by ganiuszka** [comment]
Hello @UltraBlackLinux,

Many thanks for reporting this problem.

I think I know this case. It happens: **IF** on the systems is SELinux enabled and **IF** Docker is installed using the Docker way from Docker repository in https://download.docker.com/linux/. Both conditions needs to be fullfiled to reproduce this problem.

Could I ask you about confirming if it is your case? Or it does not describe your Docker environment?

In this configuration the Bacula daemons are not able to start listening by long time. It is also the reason why you see the error message on the Bacularis web interface.

So far I couldn't find any easy solution for that. I found that installing the Docker in Moby Engine version works with these container images, while installing from docker.com does not solve it.

So in summary, it looks to be a SELinux related problem somewhere on the Docker level. We will try to find a solution for that soon. I will keep this ticket open until this time.

Alternatively if you are interested in, you can try one of the other Bacularis installation ways. They are described here:

https://bacularis.app/doc/brief/before-you-start.html

In case problems, please let us know. Good luck!

Best regards,
Marcin Haba (gani)

**Comment by ganiuszka** [comment]
Hello @UltraBlackLinux,

I researched a bit this issue. I found that Bacula has a problem with very long starting if the limit of opened files is set to high value. For Docker installed from docker.com repos this limit is set to 1073741816 which causes a problem for Bacula. I just opened a bug report in Bacula GitLab and I described it there in details:

https://gitlab.bacula.org/bacula-community-edition/bacula-community/-/issues/2716

As a workaround you can set the open file limit for the Docker containers to low value like 1024. It is possible to do by editing Docker systemd unit and adding there a parameter to set this limit.

In my system the Docker Systemd unit is here:
```
/usr/lib/systemd/system/docker.service
```

The new ``dockerd`` parameter is:

```
--default-ulimit nofile=1024:1024
```

and you can add this parameter to ``ExecStart`` line. On my side it looks now like this:

```
ExecStart=/usr/bin/dockerd --default-ulimit nofile=1024:1024 -H fd:// --containerd=/run/containerd/containerd.sock
```

After this modification, please reload the Systemd and restart the Docker:

```
# systemctl daemon-reload
# systemctl restart docker
```

Now you can try to start the ``bacularis-standalone`` container image.

Please let us know if it helped. Thanks!

Best regards,
Marcin Haba (gani)

**Comment by UltraBlackLinux** [comment]
> Could I ask you about confirming if it is your case? Or it does not describe your Docker environment?

I just checked and it doesn't appear that I have selinux installed at all. Might be worth mentioning that I'm on base arch with almost nothing else installed besides docker.

> set the open file limit for the Docker containers to low value like 1024

That seems to have worked, thank you so much!

**Comment by ganiuszka** [comment]
Hello @UltraBlackLinux

Many thanks for response. I am glad that decreasing the open file limit solved this problem.

Yes, this SELinux was a mistake. I had to do something incorrect during previous tests.

I prepared in the ``Troubleshooting`` chapter in the Bacularis documentation a section about this case:

https://bacularis.app/doc/brief/troubleshooting.html#error-about-a-problem-with-connection-to-bconsole-when-using-docker-image

I hope it will help Bacularis users in the future to solve it easier.

I am closing this issue. Thanks!

Best regards,
Marcin Haba (gani)
